scipy/stats/_rvs_sampling.py

```diff
@@ -168,3 +172,393 @@ def rvs_ratio_uniforms(pdf, umax, vmin, vmax, size=1, c=0, random_state=None):
+    '0.0911 ms'  # may vary
+
+    Depending on the implementation of the distribution's random sampling
+    method, the random variates generated may be nearly identical, given
+    the same random state.
+
+    >>> # `seed` ensures identical random streams are used by each `rvs` method
+    >>> seed = 500072020
+    >>> rvs1 = dist.rvs(size=100, random_state=np.random.default_rng(seed))
+    >>> rvs2 = fni.rvs(size=100, random_state=np.random.default_rng(seed))
+    >>> np.allclose(rvs1, rvs2)
+    True
+
+    """
+
+    def __init__(self, dist, tol=1e-12, max_intervals=100000):
+        res =_fast_numerical_inverse(dist, tol, max_intervals)
+        H, eu, intervals, a, b = res
+        self.H = H
+        self.midpoint_error = eu
+        self.intervals = intervals
+        self._a, self._b = a, b
+
+    def ppf(self, q):
+        r"""
+        Approximate percent point function (inverse `cdf`) of the given RV.
+
+        Parameters
+        ----------
+        q : array_like
+            lower tail probability.
+
+        Returns
+        -------
+        x : array_like
+            quantile corresponding to the lower tail probability `q`.
+
+        """
+        q = np.asarray(q)  # no harm; self.H always returns an array
+        result = np.zeros_like(q, dtype=np.float64)
+        i = (q >= 0) & (q <= 1)
+        result[i] = self.H(q[i])
+        result[~i] = np.nan
+        return result
+
+    def rvs(self, size=None, random_state=None):
+        """
+        Random variates of the given RV.
+
+        The `random_state` is used to draw uniform pseudo-random variates, and
+        these are converted to pseudo-random variates of the given RV using
+        inverse transform sampling.
+
+        Parameters
+        ----------
+        size : int, tuple of ints, or None; optional
+            Defines shape of array of random variates. Default is ``None``.
+        random_state : {None, int, `numpy.random.Generator`,
+                        `numpy.random.RandomState`}, optional
+
+            Defines the object to use for drawing pseudorandom variates.
+            If `random_state` is ``None`` the `np.random.RandomState`
+            singleton is used.
+            If `random_state` is an ``int``, a new ``RandomState`` instance is
+            used, seeded with `random_state`.
+            If `random_state` is already a ``RandomState`` or ``Generator``
+            instance, then that object is used.
+            Default is None.
+
+        Returns
+        -------
+        rvs : ndarray or scalar
+            Random variates of given `size`. If `size` is ``None``, a scalar
+            is returned.
+        """
+        random_state = check_random_state(random_state)
+        uniform = random_state.uniform(size=size)
+        # scale to valid domain of interpolant
+        uniform = self._a + uniform * (self._b - self._a)
+        return self.ppf(uniform)
+
+    def qrvs(self, size=None, d=None, qmc_engine=None):
```

Comment:
Maybe this needs `random_state` as well. This would be on par with `rvs` and is required if the engine is not passed.

Reply:
Why? To pass to the `Halton` constructor when `qmc_engine is None`? I'd suggest rather that if the user wants to control that, they should just construct and pass in a `Halton` instance.

Reply:
👌 then if this behavior is fine. This was my last comment I think so approving!